pinc/Round.inc

```diff
@@ -563,3 +563,61 @@ function is_formatting_round($round)
 {
     return startswith($round->id, 'F');
 }
+
+/**
+ * From an array & key, get a Round object matching the ID
+ *
+ * @param array $arr
+ * @param string $key
+ * @param mixed $default
+ * @param bool $allownull
+ *
+ * @throws InvalidArgumentException
+ */
+function get_round_param($arr, $key, $default = null, $allownull = false)
+{
+    global $Round_for_round_id_;
+    // sanity checks on the args
+    assert(is_array($arr));
+    assert(!is_array($key));
+    assert(is_null($default) || is_string($default));
+    if (!is_null($default)) {
+        assert(!$allownull);
+    }
+    if (isset($arr[$key])) {
+        $s = $arr[$key];
+        if (!is_string($s)) {
+            throw new InvalidArgumentException(sprintf(
+                _("Parameter '%1\$s' is not a valid type"),
+                $key
+            ));
+        }
+        // Trim whitespace from both ends of the string.
+        $s = trim($s);
+        if ($s == '' && $allownull) {
+            return null;
```

Comment:
This is the first place we know that `$s` is guaranteed to be a string after the `is_string()` and if we ran `trim()` against it before this it would likely coerce the variable into a string if it was a number prior.

Reply:
Thanks for the explanation -- that makes sense.